core/src/main/java/com/homeaway/streamingplatform/db/dao/impl/ConsumerDaoImpl.java

```diff
@@ -188,16 +187,20 @@ private void deleteConsumer(String streamName, String consumerName) {
         Optional<AvroStream> avroStream = getAvroStreamKeyValue(streamName).getValue();
 
         if (avroStream.isPresent()) {
-            List<com.homeaway.digitalplatform.streamregistry.Consumer> listConsumer = avroStream.get().getConsumers();
-            int consumerInitialSize = listConsumer.size();
-            for (Iterator<com.homeaway.digitalplatform.streamregistry.Consumer> iter = listConsumer.listIterator(); iter.hasNext();) {
-                com.homeaway.digitalplatform.streamregistry.Consumer consumerEntity = iter.next();
-                if (consumerEntity.getActor().getName().equalsIgnoreCase(consumerName)) {
-                    // Remove the consumer
-                    iter.remove();
-                    avroStream.get().setConsumers(Lists.newArrayList(iter));
-                }
-            }
+            // Obtains consumer list size before  remove consumer
+            final int consumerInitialSize = avroStream.get().getConsumers().size();
+
+            // Obtains filtered consumer list not containing the consumer we want to remove
+            List<com.homeaway.digitalplatform.streamregistry.Consumer> withoutConsumer = avroStream.get()
+                    .getConsumers()
```

Comment:
`avroStream.get().getConsumers()` can be refactored to a variable since it's used twice

Reply:
It's ready